ui/panels-plugin/src/plugins/line-chart/LineChart.tsx

```diff
@@ -33,51 +59,66 @@ export interface LineChartProps {
 function LineChart(props: LineChartProps) {
   const { width, height } = props;
   const queries = useRunningGraphQueries();
+  const [tooltipData, setTooltipData] = useState<TooltipData>(emptyTooltipData);
 
   // Calculate the LineChart options based on the query results
-  const option: EChartsOption = useMemo(() => {
-    const dataset: EChartsOption['dataset'] = [];
+  const { option, timeScale } = useMemo(() => {
+    const timeScale = getCommonTimeScale(queries);
+    if (timeScale === undefined) {
+      return { option: { series: undefined }, timeScale: undefined };
+    }
+
     const series: EChartsOption['series'] = [];
 
     for (const query of queries) {
       // Skip queries that are still loading and don't have data
       if (query.loading || query.data === undefined) continue;
 
-      // For every series that comes back from a query, add a Dataset and a Series
-      // to the chart
       for (const dataSeries of query.data.series) {
-        const id = dataset.length;
-
-        dataset.push({
-          id,
-          source: [['timestamp', 'value'], ...dataSeries.values],
-        });
-
         series.push({
           type: 'line',
-          datasetId: id,
           name: dataSeries.name,
+          data: [...dataSeries.values],
```

Comment:
Did you say that `dataset` has performance problems? Can't remember why you needed to do it this way.

Reply:
Yeah I usually prefer using dataset--for my [old PR with category axis type](https://github.com/perses/perses/pull/227), series data was faster than dataset, so I kept using that approach as I refactored to this version. I ran another comparison yesterday with this version (time xAxis type) and there wasn't a difference in performance. 

I'm leaning toward leaving it as is though so I don't have to refactor getNearbySeries to take another param (would have to pass option.dataset as well as option.series). Also, storing data directly with the rest of the series properties will make tooltip drill-down / chart details panel easier to implement in the future.